packages/astro/src/core/util.ts

```diff
@@ -221,3 +224,31 @@ export function resolvePath(specifier: string, importer: string) {
 		return specifier;
 	}
 }
+
+// EndpointOutput.body is typed as "Body" (not exported) which is only allowed to be a string
+export function isEndpointOutput(o: any): o is EndpointOutput {
```

Comment:
nit: I think we can give a better name instead of `o`. Maybe "responseOutput" works better